bgapi/api.py

```diff
@@ -519,109 +519,126 @@ def parse_bgapi_event(self, packet_class, packet_command, rx_payload, callbacks=
-                callbacks.ble_evt_sm_bonding_fail(handle=handle, result=result)
+                callbacks.ble_evt('ble_evt_sm_bonding_fail', handle=handle, result=result)
             elif packet_command == 2:
                 handle, passkey = struct.unpack('<BI', rx_payload[:5])
-                callbacks.ble_evt_sm_passkey_display(handle=handle, passkey=passkey)
+                callbacks.ble_evt('ble_evt_sm_passkey_display', handle=handle, passkey=passkey)
             elif packet_command == 3:
                 handle = struct.unpack('<B', rx_payload[:1])[0]
-                callbacks.ble_evt_sm_passkey_request(handle=handle)
+                callbacks.ble_evt('ble_evt_sm_passkey_request', handle=handle)
             elif packet_command == 4:
                 bond, keysize, mitm, keys = struct.unpack('<BBBB', rx_payload[:4])
-                callbacks.ble_evt_sm_bond_status(bond=bond, keysize=keysize, mitm=mitm, keys=keys)
+                callbacks.ble_evt('ble_evt_sm_bond_status', bond=bond, keysize=keysize, mitm=mitm, keys=keys)
         elif packet_class == 6:
             if packet_command == 0:
                 rssi, packet_type, sender, address_type, bond, data_len = struct.unpack('<bB6sBBB', rx_payload[:11])
-                callbacks.ble_evt_gap_scan_response(rssi=rssi, packet_type=packet_type, sender=sender,
+                callbacks.ble_evt('ble_evt_gap_scan_response', rssi=rssi, packet_type=packet_type, sender=sender,
                                                     address_type=address_type, bond=bond, data=rx_payload[11:] )
             elif packet_command == 1:
                 discover, connect = struct.unpack('<BB', rx_payload[:2])
-                callbacks.ble_evt_gap_mode_changed(discover=discover, connect=connect)
+                callbacks.ble_evt('ble_evt_gap_mode_changed', discover=discover, connect=connect)
         elif packet_class == 7:
             if packet_command == 0:
                 timestamp, port, irq, state = struct.unpack('<IBBB', rx_payload[:7])
-                callbacks.ble_evt_hardware_io_port_status(timestamp=timestamp, port=port, irq=irq, state=state)
+                callbacks.ble_evt('ble_evt_hardware_io_port_status', timestamp=timestamp, port=port, irq=irq, state=state)
             elif packet_command == 1:
                 handle = struct.unpack('<B', rx_payload[:1])[0]
-                callbacks.ble_evt_hardware_soft_timer(handle=handle)
+                callbacks.ble_evt('ble_evt_hardware_soft_timer', handle=handle)
             elif packet_command == 2:
                 input, value = struct.unpack('<Bh', rx_payload[:3])
-                callbacks.ble_evt_hardware_adc_result(input=input, value=value)
+                callbacks.ble_evt('ble_evt_hardware_adc_result', input=input, value=value)
 
-class BlueGigaCallbacks(object):
+
+class BlueGigaCallbacksBase(object):
+    def ble_rsp(self, response, **kargs):
+        callback = getattr(self, response, None)
```

Comment:
Hi @pinkavaj , I know I haven't been involved in the project much so I apologize for jumping in unexpectedly, but I have some confusion about this PR. In what cases would `getattr` return None? If the member doesn't exist on the `BlueGigaCallbacks` class then shouldn't it be added and called explicitly?